distributed/scheduler.py

```diff
@@ -2167,6 +2169,12 @@ def add_plugin(self, plugin=None, idempotent=False, **kwargs):
 
         self.plugins.append(plugin)
 
+        if self.status == "running":
+            try:
+                plugin.restart(self)
+            except Exception as e:
+                logger.exception(e)
```

Comment:
I'm not sure we want to trigger the `restart` event for plugins on startup.  I'm not sure how expected this would be by everyone.  Instead I recommend providing the scheduler as an argument as you create the plugin.

Reply:
I may not be fully understanding the intent here though.

Reply:
The plugin (as currently designed) wants to make a Client. The Client requires a running Scheduler. Typically you add a plugin before the scheduler is running... so you need some way to know when the scheduler started.

There might be some other approach.